tests/baselines/reference/useBeforeDeclaration_destructuring.types

```diff
@@ -0,0 +1,23 @@
+=== tests/cases/compiler/useBeforeDeclaration_destructuring.ts ===
+a;
+>a : any
+
+let {a, b = a} = {a: '', b: 1};
+>a : any
+>b : any
+>a : any
+>{a: '', b: 1} : { a: string; b?: number; }
+>a : string
+>'' : ""
+>b : number
+>1 : 1
+
+b;
+>b : any

```

Comment:
seems like there needs to be a change to the control flow as well to correctly infer `b: string | number`